src/Mod/PartDesign/Gui/TaskFeatureParameters.cpp

```diff
@@ -191,12 +196,23 @@ bool TaskDlgFeatureParameters::accept()
         Gui::cmdGuiDocument(feature, "resetEdit()");
         Gui::Command::commitCommand();
     } catch (const Base::Exception& e) {
-        // Generally the only thing that should fail is feature->isValid() others should be fine
-        QString errorText = QApplication::translate(feature->getTypeId().getName(), e.what());
-        QMessageBox::warning(Gui::getMainWindow(), tr("Input error"), errorText);
-        return false;
-    }
 
+      QString errorText = QString::fromUtf8(e.what());
+      QString statusText = QString::fromUtf8(getObject()->getStatusString());
+
+      // generic, fallback error message
+      if (errorText == "Error" || errorText.isEmpty()) {
+        if (!statusText.isEmpty() && statusText != "Error") {
+          errorText = statusText;
+        } else {
+          errorText = tr("The feature could not be created with the given parameters.\n"
+              "The geometry may be invalid or the parameters may be incompatible.\n"
+              "Please adjust the parameters and try again.");
```

Comment:
@kaktusus heads-up to translators that the nature of this bug effectively requires a string addition to resolve it, so I plan on merging despite the current string freeze.